cvmfs/util/posix.cc

```diff
@@ -540,8 +543,11 @@ void ReadHalfPipe(int fd, void *buf, size_t nbyte) {
       SafeSleepMs(backoff_ms);
       if (backoff_ms < max_backoff_ms) backoff_ms *= 2;
     }
+
+  duration_ms = (platform_monotonic_time_ns() - timestamp) / 1000;
+  if (timeout_ms != 0 && duration_ms  > timeout_ms) return false;
   } while (num_bytes == 0);
-  assert((num_bytes >= 0) && (static_cast<size_t>(num_bytes) == nbyte));
+  return ((num_bytes >= 0) && (static_cast<size_t>(num_bytes) == nbyte));
```

Comment:
I think we should leave the assert. At this point, we should really have all the bytes read.